src/tools/pkcs11-tool.c

```diff
@@ -6940,249 +7012,249 @@ static void pseudo_randomize(unsigned char *data, size_t dataLen)
+      { CKM_JUNIPER_WRAP,	"JUNIPER-WRAP", NULL, MF_UNKNOWN },
+      { CKM_FASTHASH,		"FASTHASH", NULL, MF_UNKNOWN },
+      { CKM_AES_KEY_GEN,	"AES-KEY-GEN", NULL, MF_UNKNOWN },
+      { CKM_AES_ECB,		"AES-ECB", NULL, MF_UNKNOWN },
+      { CKM_AES_CBC,		"AES-CBC", NULL, MF_UNKNOWN },
+      { CKM_AES_MAC,		"AES-MAC", NULL, MF_UNKNOWN },
+      { CKM_AES_MAC_GENERAL,	"AES-MAC-GENERAL", NULL, MF_UNKNOWN },
+      { CKM_AES_CBC_PAD,	"AES-CBC-PAD", NULL, MF_UNKNOWN },
+      { CKM_AES_CTR,		"AES-CTR", NULL, MF_UNKNOWN },
+      { CKM_AES_GCM,		"AES-GCM", NULL, MF_UNKNOWN },
+      { CKM_AES_CMAC,		"AES-CMAC", NULL, MF_UNKNOWN },
+      { CKM_DES_ECB_ENCRYPT_DATA, "DES-ECB-ENCRYPT-DATA", NULL, MF_UNKNOWN },
+      { CKM_DES_CBC_ENCRYPT_DATA, "DES-CBC-ENCRYPT-DATA", NULL, MF_UNKNOWN },
+      { CKM_DES3_ECB_ENCRYPT_DATA, "DES3-ECB-ENCRYPT-DATA", NULL, MF_UNKNOWN },
+      { CKM_DES3_CBC_ENCRYPT_DATA, "DES3-CBC-ENCRYPT-DATA", NULL, MF_UNKNOWN },
+      { CKM_AES_ECB_ENCRYPT_DATA, "AES-ECB-ENCRYPT-DATA", NULL, MF_UNKNOWN },
+      { CKM_AES_CBC_ENCRYPT_DATA, "AES-CBC-ENCRYPT-DATA", NULL, MF_UNKNOWN },
+      { CKM_GOST28147_KEY_GEN,	"GOST28147-KEY-GEN", NULL, MF_UNKNOWN },
+      { CKM_GOST28147_ECB,	"GOST28147-ECB", NULL, MF_UNKNOWN },
+      { CKM_GOST28147,	"GOST28147", NULL, MF_UNKNOWN },
+      { CKM_GOST28147_MAC,	"GOST28147-MAC", NULL, MF_UNKNOWN },
+      { CKM_GOST28147_KEY_WRAP,	"GOST28147-KEY-WRAP", NULL, MF_UNKNOWN },
+      { CKM_GOSTR3410_KEY_PAIR_GEN,"GOSTR3410-KEY-PAIR-GEN", NULL, MF_UNKNOWN },
+      { CKM_GOSTR3410,		"GOSTR3410", NULL, MF_UNKNOWN },
+      { CKM_GOSTR3410_DERIVE,	"GOSTR3410-DERIVE", NULL, MF_UNKNOWN },
+      { CKM_GOSTR3410_WITH_GOSTR3411,"GOSTR3410-WITH-GOSTR3411", NULL, MF_UNKNOWN },
+      { CKM_GOSTR3410_512_KEY_PAIR_GEN,	"GOSTR3410-512-KEY-PAIR-GEN", NULL, MF_UNKNOWN },
+      { CKM_GOSTR3410_512,	"GOSTR3410_512", NULL, MF_UNKNOWN },
+      { CKM_GOSTR3410_12_DERIVE,	"GOSTR3410-12-DERIVE", NULL, MF_UNKNOWN },
+      { CKM_GOSTR3410_WITH_GOSTR3411_12_256,	"GOSTR3410-WITH-GOSTR3411-12-256", NULL, MF_UNKNOWN },
+      { CKM_GOSTR3410_WITH_GOSTR3411_12_512,	"GOSTR3410-WITH-GOSTR3411-12-512", NULL, MF_UNKNOWN },
+      { CKM_GOSTR3411,		"GOSTR3411", NULL, MF_UNKNOWN },
+      { CKM_GOSTR3411_HMAC,	"GOSTR3411-HMAC", NULL, MF_UNKNOWN },
+      { CKM_GOSTR3411_12_256,	"GOSTR3411-12-256", NULL, MF_UNKNOWN },
+      { CKM_GOSTR3411_12_512,	"GOSTR3411-12-512", NULL, MF_UNKNOWN },
+      { CKM_GOSTR3411_12_256_HMAC,	"GOSTR3411-12-256-HMAC", NULL, MF_UNKNOWN },
+      { CKM_GOSTR3411_12_512_HMAC,	"GOSTR3411-12-512-HMAC", NULL, MF_UNKNOWN },
+      { CKM_DSA_PARAMETER_GEN,	"DSA-PARAMETER-GEN", NULL, MF_UNKNOWN },
+      { CKM_DH_PKCS_PARAMETER_GEN,"DH-PKCS-PARAMETER-GEN", NULL, MF_UNKNOWN },
+      { CKM_X9_42_DH_PARAMETER_GEN,"X9-42-DH-PARAMETER-GEN", NULL, MF_UNKNOWN },
+      { CKM_AES_KEY_WRAP,	"AES-KEY-WRAP", NULL, MF_UNKNOWN },
+      { 0, NULL, NULL, MF_UNKNOWN }
```

Comment:
The `MF_UNKNOWN` is 0 so if we would skip this initializer, it would be the same, isn't it? It would also make it clear what entry is already filled in.

Reply:
I'm not getting your suggestion here, are you saying you want:
```C
      { 0, NULL, NULL, 0 }
```
The existing code has this as a NULL terminated array, not sure it really needs to be NULL term since size is known.

Reply:
This was comment for the whole block, not only for the terminator. My suggestion was to drop the MF_UNKNOWN from all the "incomplete" entries that are not yet filled out with sensible flags. But on second though, I am not sure if it would be better.

Reply:
> This was comment for the whole block, not only for the terminator. My suggestion was to drop the MF_UNKNOWN from all the "incomplete" entries that are not yet filled out with sensible flags. But on second though, I am not sure if it would be better.

You have to put an initializer there do to the default compiler flags (-Werror=missing-field-initializers):
```
pkcs11-tool.c:7247:7: error: missing initializer for field ‘mf_flags’ of ‘struct mech_info’ [-Werror=missing-field-initializers]
 7247 |       { CKM_AES_KEY_WRAP, "AES-KEY-WRAP", NULL },
```
So I thought something descriptive, like UNKNOWN would be better than a 0.

Reply:
Oh, right. My bad. So then it is fine, I guess.